jaxrs-api/src/main/java/javax/ws/rs/JAXRS.java

```diff
@@ -0,0 +1,375 @@
+         * <p>
+         * Same as if calling {@link #property(String) property(PROTOCOL)}.
+         * </p>
+         *
+         * @return protocol to be used (e. g. {@code "HTTP")}.
+         * @throws ClassCastException
+         *             if protocol is not a {@link String}.
+         * @see JAXRS.Configuration#PROTOCOL
+         */
+        default String protocol() {
+            return (String) property(PROTOCOL);
+        }
+
+        /**
+         * Convenience method to get the {@code host} to be used.
+         * <p>
+         * Same as if calling {@link #property(String) (String) property(HOST)}.
+         * </p>
+         *
+         * @return host name or IP address to be used (e. g. {@code "localhost"} or
+         *         {@code "0.0.0.0"}).
+         * @throws ClassCastException
+         *             if host is not a {@link String}.
+         * @see JAXRS.Configuration#HOST
+         */
+        default String host() {
+            return (String) property(HOST);
+        }
+
+        /**
+         * Convenience method to get the {@code port} to be used.
+         * <p>
+         * Same as if calling {@link #property(String) property(PORT)}.
+         * </p>
+         * <p>
+         * If the port was not explicitly given, this will return the default port
+         * chosen by the JAX-RS implementation.
+         * </p>
+         *
+         * @return port number to be used (e. g. {@code 8080}.
+         * @throws ClassCastException
+         *             if port is not an {@code int}.
+         * @see JAXRS.Configuration#PORT
+         */
+        default int port() {
+            return (int) property(PORT);
+        }
+
+        /**
+         * Convenience method to get the {@code rootPath} to be used.
+         * <p>
+         * Same as if calling {@link #property(String) property(ROOT_PATH)}.
+         * </p>
+         *
+         * @return root path to be used, e. g. {@code "/"}.
+         * @throws ClassCastException
+         *             if root path is not a {@link String}.
+         * @see JAXRS.Configuration#ROOT_PATH
+         */
+        default String rootPath() {
+            return (String) property(ROOT_PATH);
+        }
+
+        /**
+         * @return {@link Builder} for bootstrap configuration.
+         */
+        static Builder builder() {
+            return RuntimeDelegate.getInstance().createConfigurationBuilder();
+        };
+
+        /**
+         * Builder for bootstrap {@link Configuration}.
+         *
+         * @author Markus KARG (markus@headcrashing.eu)
+         *
+         * @since 2.2
+         */
+        static interface Builder {
+
+            /**
+             * @return {@link Configuration} built from provided property values.
+             */
+            Configuration build();
+
+            /**
+             * Sets the property {@code name} to the provided {@code value}.
+             * <p>
+             * This method ignores unknown names and invalid values.
```

Comment:
> This method ignores .... invalid values.

This sounds like you won't get any error for calling something like `.property(HOST, BigDecimal.ZERO)`. Is that correct?

Reply:
Yes, the API does not mandate a compliant implementation to check for validity of values this point in time. The reason is that when I coded the RI ontop of Jersey I noticed that for some properties it is not clear what a "valid" value is at that time, because of dependencies between the properties. For example, `protocol("HTTPS")` enforces TLS; so they need a `SSLContext`. But at the time of calling `protocol("HTTPS")` it is not set, so we cannot check it *here*. Some backends do such complex checks only *very very late* (I had to debug deep into Netty to learn it) and it makes no sense to have only 10% checked right here, while 90% are checked at time of actual server startup. Hence I decided not to demand such checks *here*. I thought about *build()* to do that, but even that would not work with Jersey+Netty, at least without substantial rewrite of Netty, Jetty, Grizzly, and so on to get publicly available checks *before* the actual server boot.

Reply:
I'm not sure if I fully understand. 

Let assume the user calls `.property(HOST, BigDecimal.ZERO)` or `.property(HOST, "I love JAX-RS")` and starts the server afterwards. What's the expected behavior? I would expect the bootstrapping to fail with some error like "Invalid hostname/IP". 

But from the current wording the invalid value is **ignored** by the `.property()` method. If the value is ignored, the configuration would still be configured to use the default `0.0.0.0` and the server will bootstrap without any error.

I hope this makes my point clearer. I think the wording "ignore invalid value" is a bit confusing.

Reply:
I see your point. I should rephrase the Javadocs to say this: `This method does not check the validity, type or syntax of the provided value`.

Reply:
Great! This wording is much better!